library/L1_Peripheral/lpc40xx/watchdog_interface.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstdint>
+#include <iterator>
+
+namespace sjsu
+{
+
+class WatchdogInterface
+{
+ public:
+     virtual void Initialize(uint32_t timer_constant, uint32_t timer_window,
+     uint32_t timer_warning) = 0;
+     virtual void RegisterIrqHandler() = 0;
+     virtual void FeedSequence() = 0;
+     virtual uint32_t CheckTimeout() = 0;
+};

```

Comment:
Maybe too much for this commit, (meaning do not implement this now), but it would be great if we could have the watchdog reset the board it is triggered.